docs/modules/ROOT/pages/how-tos/configuring/customizing-the-build.adoc

```diff
@@ -1 +1,168 @@
+. Commit your changes to the repository of the component.
+
+
+== Extending the build pipeline with your own tasks
+
+Before you extend your build pipeline, be aware that doing so causes builds to fail an link:https://enterprisecontract.dev/docs/ec/main/index.html[Enterprise Contract] check, if you are using the EC. To resolve this issue, please reference the next procedure after completing this one.  
+
+.Procedure
+
+. In each `.yaml` file in the `.tekton` directory, add a new task under `tasks`. 
+
++ 
+Example task:
++
+[source]
+--
+  name: example-task
+  params:
+  - name: example-param
+    value: “Example”
+  runAfter:
+  - build-container #You can be more specific by choosing another task
+  taskRef:
+    params:
+    - name: name
+      value: example-task # metadata.name field of the Task
+    - name: bundle
+      value: quay.io/tekton-bundle-catalog/example-task-bundle:1.0
+      # For more details on tekton bundles, refer https://tekton.dev/docs/pipelines/pipelines/#tekton-bundles
+    - name: kind
+      value: task
+    resolver: bundles
+  when:
+  - input: $(params.skip-checks)    #This references the pipeline parameters
+    operator: in
+    values:
+    - "false"
+--
+
++
+An example of a custom task added to the pipeline that sends a slack notification when the `Pipelinerun` fails:
++
+[source]
+--
+  name: slack-webhook-notification
+  params:
+    - name: message
+      value: PipelineRun $(context.pipelineRun.name) failed
+    - name: secret-name
+      value: my-secret # name of secret in the your namespace which contains slack web-hook URL under key specified in 'key-name' parameter below
+    - name: key-name
+      value: dev-team
+  taskRef:
+    params:
+    - name: bundle
+      value: quay.io/redhat-appstudio-tekton-catalog/task-slack-webhook-notification:0.1
+    - name: name
+      value: slack-webhook-notification
+    - name: kind
+      value: Task
+    resolver: bundles
+  when:
+    - input: $(tasks.status)
+      operator: in
+      values: ["Failed"]
+--
+
+. Commit your changes to the repository of the component.
+
+[NOTE] 
+====
+* To use `slack-webhook-notification` task, you need to create a secret in your namespace with at least one key where the value is the webhook URL. For example, to create a secret for Slack, run `oc create secret generic my-secret --from-literal dev-team=https://hooks.slack.com/services/XXX/XXXXXX`
+
+* If you want to define a task directly in this file, rather than using `taskRef`, you can use `taskSpec`. Visit the documentation linked in the <<additional-resources>> section.
+
+====
+
+== Preventing issues with the Enterprise Contract
+
+If you are using the Enterprise Contract (EC) to verify your builds, then extending your build pipeline causes your builds to fail the EC. Specifically, builds begin to violate the "trusted tasks" rule.  
+
+At the time of publication, the easiest solution for this problem we can recommend is to skip the trusted tasks check. 
```

Comment:
Can we link to the specific topic, i.e. https://enterprisecontract.dev/docs/ec-policies/release_policy.html#tasks__required_untrusted_task_found

@lcarva @simonbaird , is this the proper reference? Is this link stable?

Reply:
I did talk with Simon about this yesterday as well, and he pointed me to this topic, which I'll include in the commit I'm working on: https://enterprisecontract.dev/docs/ec-policies/release_policy.html#trusted_task__trusted